apps/researcher/src/lib/user/actions.ts

```diff
@@ -0,0 +1,35 @@
+'use server';
+
+import {clerkClient} from '@clerk/nextjs';
+import {unstable_noStore as noStore} from 'next/cache';
+
+interface AddAttributionIdProps {
+  userId: string;
+  attributionId: string;
+}
+
+export async function addAttributionId({
+  userId,
+  attributionId,
+}: AddAttributionIdProps) {
+  noStore();
+
+  if (!userId) {
+    throw new Error('userId is required');
+  }
+
+  const user = await clerkClient.users.getUser(userId);
+
+  const existingAttributionIds =
+    (user.publicMetadata?.attributionIds as string[]) || [];
+
+  const newAttributionIds = Array.from(
+    new Set([...existingAttributionIds, attributionId])
+  );
```

Comment:
Perhaps add a comment on top of these lines that makes clears why these actions are necessary?